src/models/event.ts

```diff
@@ -262,6 +262,16 @@ export class MatrixEvent extends EventEmitter {
         this.localTimestamp = Date.now() - this.getAge();
     }
 
+    /**
+     * Gets the event as though it would appear unencrypted. If the event is already not
+     * encrypted, it is simply returned as-is.
+     * @returns {IEvent} The event in wire format.
+     */
+    public getEffectiveEvent(): IEvent {
+        // clearEvent doesn't have all the fields, so we'll copy what we can from this.event
+        return Object.assign({}, this.event, this.clearEvent) as IEvent;
```

Comment:
are you allergic to the nicer syntax :P?

```
return {
    ...this.event,
    ...this.clearEvent,
}
```

Reply:
"nicer" - I feel it doesn't communicate the order properly given JS objects are unordered, so there's no implicit guarantee that the spread operation will be applied correctly. The spec says it will, but that's a detail someone has to remember when writing/reading the code.